### ISSUE TITLE
Add ARM64 CI tests

### ISSUE THREAD

**Issue by odidev** [open-telemetry/opentelemetry-collector-contrib#12920]
Hi Team, 

I'm interested in adding arm64 support in the ci to this repository. 

**Is your feature request related to a problem? Please describe.**
This repository doesn’t have arm64 testing support in the GitHub actions ci. 

**Describe the solution you'd like**
I have explored the repository and CI being used, arm64 builds can be added in the GitHub actions CI to enable test for arm64. 

**Describe alternatives you've considered**
I am working to add arm64 support in GitHub actions to add builds for testing, we can do ARM64 testing in GitHub actions using self-hosted arm64 runners. 

If interested, we can get access to ARM H/W through [WorksOnArm](https://github.com/WorksOnArm).

**Additional context**
Do you have any plans for adding the arm64 platform in CI? It will be very helpful if arm64 support is available. 

Please share your thoughts on this.

**Comment by mx-psi** [comment]
@odidev Thanks for the request!

Note that we already ensure the Collector builds on ARM through cross-compilation, so I assume this is about unit tests? Or did you have a different use case in mind?

> I have explored the repository and CI being used, arm64 builds can be added in the GitHub actions CI to enable test for arm64.

How can this be done? I tried reading actions/virtual-environments/issues/2552 but found it a bit confusing and I am not sure that this is natively supported by Github. Could you share a bit more about this?

>  Do you have any plans for adding the arm64 platform in CI? It will be very helpful if arm64 support is available.

AFAIK there are not plans for adding this. We do not have a well-defined policy on what level of testing and support we provide for different architectures, so it's hard to give a general answer that reflects general consensus.

My personal opinion is that adding this would be okay so long as
1. we track and actively fix or skip unit tests failing on ARM
2. this does not increase the running time of CI for PRs too much

**Comment by brettmc** [comment]
We have a [lowish-priority issue](https://github.com/open-telemetry/opentelemetry-php/issues/904) over in opentelemetry-php to try building/testing against arm64, and having access to an arm-based self-hosted runner seems like it would help us out.

At the moment, our options seem to be using virtualization, or waiting for github to add arm support to their runners.

**Comment by atoulme** [comment]
I am in talks with a vendor to get an appliance to run arm tests as a github action runners. I will assign this issue to myself.

**Comment by dneary** [comment]
I would love to help get this to happen! How can I help? What would be useful - hardware, cloud credits, or something else? Doing native builds and tests should speed up builds.

**Comment by atoulme** [comment]
I have an Ampere Mt Collins server provided for free by Ampere Computing to support this. It's sitting in my garage and I need help to set it up to use different VMs for different github action workers. Please allow me some time to get to it.

**Comment by dneary** [comment]
> I have an Ampere Mt Collins server provided for free by Ampere Computing to support this. It's sitting in my garage and I need help to set it up to use different VMs for different github action workers. Please allow me some time to get to it.

Antoine, we may have an even better option! I believe you already spoke to my colleague Peter - early next week, we should be able to confirm your participation in the actuated pilot - the full details of the pilot, and the link for "How do we get access?" is here: [Announcing managed Arm CI for CNCF projects](https://actuated.dev/blog/arm-ci-cncf-ampere)

**Comment by atoulme** [comment]
Absolutely, I am working with Peter right now on blogging about our experience. I will read this ASAP.

**Comment by mx-psi** [comment]
@atoulme Should we close this in favor or open-telemetry/opentelemetry-collector/issues/9731?

**Comment by atoulme** [comment]
I think we can keep this issue and make it a blocker to close https://github.com/open-telemetry/opentelemetry-collector/issues/9731.

Currently, we are trying to add an arm64 CI for the collector core repository with https://github.com/open-telemetry/opentelemetry-collector/pull/9584, once that is merged, I can work to add the same level of CI support for contrib.

**Comment by mx-psi** [comment]
Alright, marking as 'on hold' then

**Comment by atoulme** [comment]
No longer on hold.

**Comment by atoulme** [comment]
This is now complete. Please feel free to review the logs and push for more ARM64 specific tests!
